lambda_backend/route_microservice/src/deleteUserDispatcher.ts

```diff
@@ -0,0 +1,64 @@
+import { Handler, SNSEvent } from 'aws-lambda';
+import DynamoDB, { AttributeValue, DeleteItemInput } from 'aws-sdk/clients/dynamodb';
+
+import { getUserRoutesIndexItems } from './common/db';
+import { logger } from './common/logger';
+
+const dynamoDb = new DynamoDB.DocumentClient();
+
+/**
+ * Listens to SNS deleteUser events and deletes corresponding routes
+ */
+export const handler: Handler = async (event: SNSEvent) => {
+  if (!process.env['ROUTE_TABLE_NAME']) {
+    // Not HTTP Error
+    throw Error('Table or Bucket names are not set!');
+  }
+  const { Records } = event;
+  const {
+    Sns: { Message: username },
+  } = Records[0];
+  logger.info('deleteUserDispatcher initiated', { data: { username } });
+
+  const userRoutesIndexItems = await getUserRoutesIndexItems(username);
+
+  logger.info('deleteUserDispatcher retrieved userRoutesIndexItems', { data: { username } });
+  for (const userRoutesIndexItem of userRoutesIndexItems) {
+    await deleteRouteWithRetry(userRoutesIndexItem.username, userRoutesIndexItem.createdAt);
```

Comment:
Should you surround this with try catch and return statusCode 400/500?

Reply:
This is just a lambda function listening to SNS. No http involved and there is alrdy logger listening for errors.

Reply:
If no HTTP is involved is there a need to return status code 200?